accessibility-checker-engine/src/v2/checker/accessibility/rules/rpt-ariaLabeling-rules.ts

```diff
@@ -20,8 +20,174 @@ import { ARIADefinitions } from "../../../aria/ARIADefinitions";
+         * AND the same role as another landmark, 
+         * AND is not differentiated by aria-label or aria-labelledby. 
+         * This causes it to be difficult for a keyboard user to know the difference between two landmarks
+         * Origin:  https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA13 this is not directly part of the ARIA spec so this is only in the IBM rules as a Violation
+         * 
+         * NOTE: When we have two landmarks at the root level of the document this rule will not check for that. 
+         * For example if we have <body> <main id="main1"></main> <main id="main2"><main> </body> we do not fail this rule. 
+         * Althought this might be an accessibility error anyway. See:
+         * https://stackoverflow.com/questions/34896476/can-i-use-more-than-one-main-html-tag-in-the-same-page/34906037
+         */
+        id: "landmark_name_unique",
+        context: "aria:complementary, aria:banner, aria:contentinfo, aria:main, aria:navigation, aria:region, aria:search, aria:form",
+        run: (context: RuleContext, options?: {}): RuleResult | RuleResult[] => {
+            // TODO do I need to fiter out bad contentinfo nodes: The footer element is not a contentinfo landmark when it is a descendant of the following HTML5 sectioning elements: https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/landmarks/HTML5.html
+            const ruleContext = context["dom"].node as Element;
+            let ownerDocument = FragmentUtil.getOwnerFragment(ruleContext);
+            let formCache = RPTUtil.getCache(ruleContext.ownerDocument, "landmark_name_unique", null);
+
+            if (!formCache) {
+                // console.log("---------ENTERING FORM CACHE")
+                formCache = {
+                    navigationNodes: [],
+                    navigationNodesComputedLabels: [],
+                    navigationNodesParents: [],
+                    navigationNodesMatchFound: []
+                }
+                let navigationNodesTemp = ownerDocument.querySelectorAll('aside,[role="complementary"], footer,[role="contentinfo"], header,[role="banner"], main,[role="main"], nav,[role="navigation"], section,[role="region"],[role="search"]');
+                let navigationNodes = Array.from(navigationNodesTemp);
+                let navigationNodesParents = [];
+                let navigationNodesMatchFound = [];
+
+                for (let i = 0; i < navigationNodes.length; i++) { // Loop over all the landmark nodes
+                    let els = [];
+                    let a = navigationNodes[i].parentElement
+                    while (a) {
+                        els.push(a);
+                        a = a.parentElement;
+                    }
+
+                    for (let j = 0; j < els.length; j++) { // Loop over all the parents of the landmark nodes
+                        // Find nearest landmark parent based on the tagName or the role attribute 
```

Comment:
Another suggestion for cleaner code:
``` ["ASIDE", "FOOTER", "FORM", ..... ].includes(els[j].tagName); ```
Makes it easier to tweak if the standard changes.

Reply:
Thanks for the idea. I went ahead and did this and made them less verbose.

Reply:
![image](https://user-images.githubusercontent.com/4451128/145294154-9c5803ca-8de6-495f-b4ed-fac2d5b6890e.png)

Reply:
The only concern on the getAttribute("role") bit, and why I didn't mention it on that line... technically, role can allow multiple values (e.g., "newrole banner"), and then the browser is expected to use the first role that it recognizes. In practice, this is pretty rare. Let's leave it as-is, but if you get a chance, a comment with // TODO wouldn't hurt.

Reply:
Gotcha ill add a todo to this line